pkg/discovery/monitoring/kubelet/kubelet_client.go

```diff
@@ -2,46 +2,44 @@ package kubelet
 
 import (
 	"fmt"
+	"github.com/golang/glog"
 	"net/http"
 	"net/url"
-
-	stats "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
-	kClient "k8s.io/kubernetes/pkg/kubelet/client"
-
-	"github.com/turbonomic/kubeturbo/pkg/discovery/util/httputil"
+	"time"
 
 	cadvisorapi "github.com/google/cadvisor/info/v1"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/util/httputil"
+	netutil "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+	stats "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
 )
 
 const (
 	summaryPath string = "/stats/summary/"
 	specPath    string = "/spec"
-)
 
-type Host struct {
-	IP       string
-	Port     int
-	Resource string
-}
+	DefaultKubeletPort  = 10255
+	DefaultKubeletHttps = false
 
-type kubeletClient struct {
-	config *kClient.KubeletClientConfig
-	client *http.Client
-}
+	defaultConnTimeOut         = 20 * time.Second
+	defaultTLSHandShakeTimeout = 10 * time.Second
+)
 
-func (kc *kubeletClient) GetPort() int {
-	return int(kc.config.Port)
+// Since http.Client is thread safe (https://golang.org/src/net/http/client.go)
+// KubeletClient is also thread-safe if concurrent goroutines won't change the fields.
+type KubeletClient struct {
+	client *http.Client
+	scheme string
+	port   int
 }
 
-func (kc *kubeletClient) GetSummary(host Host) (*stats.Summary, error) {
+func (kc *KubeletClient) GetSummary(host string) (*stats.Summary, error) {
```

Comment:
Maybe change the parameter name from `host` to `ip`, because only IP address is passed in. Same for other methods.

Reply:
It is possible to get information via hostName. I think it is more natural to use hostName in kubernetes cluster, because when deploying kubernetes, we provides a hostName list.

Reply:
Host name is provided by the cluster admin while provisioning the cluster, it might be more reliable than IP, which is derived from the host name.